### PR TITLE
Add metadata as part of metadata import

### PR DESCRIPTION
**Pull request by texasaggie97**
[X] This contribution adheres to [CONTRIBUTING.md]
### What does this Pull Request accomplish?
* Move add_metadata functions into metadata_utilities.py
* Always add additional metadata to functions as part of metadata module import
* Mako template no longer need to add the extra metadata individually
* Add unit test for adding extra metadata

### Why should this Pull Request be merged?
* Step towards consolidating helper functionality

### What testing has been done?
* Travis & System tests

### CODE REVIEW

File: build/metadata_utilities.py [hunk 1]
```diff
@@ -26,9 +31,88 @@ def merge_dicts(into, outof):
         else:
             into[item] = outof[item]
 
+    return
```
Comment (marcoskirsch) [inline build/metadata_utilities.py]: seems unecessary

File: build/metadata_utilities.py [hunk 2]
```diff
@@ -37,46 +121,108 @@ def test_merge_dict_second_is_empty():
...
+    _do_the_test_merge_dicts(a,b, expected)
 
 def test_merge_dict_with_regex():
     a        = {'aaa':{}, 'aaaa':{'x':98}, 'aaaaa':{}, 'bbb':{'bbb1':2, 'bbb2':3}, 'ccc':['x', 'y', 'z']}
     b        = {'a+':{'z':99}}
     expected = {'aaa':{'z':99}, 'aaaa':{'x':98, 'z':99}, 'aaaaa':{'z':99}, 'bbb':{'bbb1':2, 'bbb2':3}, 'ccc':['x', 'y', 'z']}
-    _do_the_test(a,b, expected)
+    _do_the_test_merge_dicts(a,b, expected)
+
+def _do_the_test_add_all_metadata(funtions, expected):
```
Comment (marcoskirsch) [inline build/metadata_utilities.py]: typo funtions

File: build/metadata_utilities.py [hunk 3]
```diff
@@ -37,46 +121,108 @@ def test_merge_dict_second_is_empty():
...
+    actual = copy.deepcopy(funtions)
```
Comment (marcoskirsch) [inline build/metadata_utilities.py]: I didn't do deep copy in line 116. Did I screw up?

File: build/metadata_utilities.py [hunk 4]
```diff
@@ -37,46 +121,108 @@ def test_merge_dict_second_is_empty():
...
+            'python_name': 'close',
+            'returns': 'ViStatus',
+            'returns_ctype': 'ViStatus_ctype',
+            'returns_python': 'ViStatus'
+        }
+    }
+
+    _do_the_test_add_all_metadata(functions, expected)
+
+
```
Comment (marcoskirsch) [inline build/metadata_utilities.py]: This is greater, no doubt better than what we had (no tests), but definitely not thorough.
We should

* Add this file to coverage
* Open issue about writing unit tests for each function

File: build/metadata_utilities.py (hunk 1 above)
Comment (texasaggie97) [reply build/metadata_utilities.py]: Removed

File: build/metadata_utilities.py (hunk 2 above)
Comment (texasaggie97) [reply build/metadata_utilities.py]: Fixed

File: build/metadata_utilities.py (hunk 3 above)
Comment (texasaggie97) [reply build/metadata_utilities.py]: I would say, if it worked, you didn't do it wrong. When I tried using a copy(), it didn't work,

In your function, is `a != actual`? When I didn't use `deepcopy()`, actual and function would end up being the same

File: build/metadata_utilities.py (hunk 4 above)
Comment (texasaggie97) [reply build/metadata_utilities.py]: We are trying to get coverage data on it, but while the tests run, it doesn't seem to be reporting coverage information.

**Pull request merged**